# Superstore Data Analysis Project

## Phase 1 — Business Understanding

### 1.1 Business Problem

Superstore generates significant sales, but management is concerned that strong revenue may be hiding profitability problems. The company wants to understand where the business is performing well, where it is losing money, and what actions could improve profitability without damaging sustainable sales growth.

### Project Goal

The goal of this project is to analyze Superstore's sales and profitability performance, identify the key factors driving profits and losses, and provide data-driven recommendations that can help management improve profitability while maintaining sustainable sales growth.

### 1.2 Stakeholders

The key stakeholders for this analysis are:

- **Executive Management:** Requires an overall view of business performance and insights to support strategic decision-making.
- **Sales Management:** Needs to understand sales performance across regions, customer segments, and product categories.
- **Finance Management:** Focuses on profitability, profit margins, losses, and the financial impact of discounts.
- **Regional Managers:** Need to identify high-performing and underperforming regions, states, and markets.
- **Product and Category Managers:** Need insights into which products and categories are driving sales and profitability and which are generating losses.
- **Marketing Management:** Can use customer and segment-level insights to improve targeting and promotional strategies.

### 1.3 Analysis Objectives

The objectives of this analysis are to:

- Evaluate the overall sales and profitability performance of Superstore.
- Analyze sales and profit trends over time to identify growth patterns and periods of underperformance.
- Compare performance across regions, states, and cities to identify strong and weak geographic markets.
- Evaluate the performance of product categories, sub-categories, and individual products.
- Identify products, locations, and customer segments that contribute significantly to profits or losses.
- Investigate the relationship between discounts and profitability.
- Analyze customer segments to understand their contribution to sales and profit.
- Identify key patterns and opportunities that could help improve business performance.
- Develop data-driven recommendations to improve profitability while supporting sustainable sales growth.
- Analyze shipping performance by examining delivery times and shipping modes.
- Analyze customer purchasing behavior and identify high-value customers.
- Distinguish between transaction line-level metrics and order-level metrics to ensure accurate calculations.

### 1.4 Analytical Business Questions

The analysis will seek to answer the following business questions:

#### Overall Business Performance
1. What are the total sales, total profit, total orders, quantity sold, average order value, and overall profit margin?
2. How have sales and profits changed over time?
3. Is Superstore achieving sales growth while maintaining profitability?

#### Geographic Performance
4. Which regions generate the highest sales and profits?
5. Which states and cities are the strongest and weakest performers?
6. Which geographic markets generate high sales but low or negative profits?

#### Product Performance
7. Which product categories and sub-categories generate the most sales and profit?
8. Which products are the most and least profitable?
9. Which products generate high sales but poor profitability?
10. Which products and sub-categories contribute most significantly to overall losses?

#### Customer and Segment Performance
11. Which customer segments generate the most sales and profit?
12. Who are Superstore's highest-value customers?
13. Which customers generate high sales but relatively low or negative profits?
14. How frequently do customers purchase from Superstore, and are there identifiable groups of high-value or at-risk customers?

#### Discount and Profitability Analysis
15. What is the relationship between discounts and profitability?
16. At what discount levels do transactions become more likely to generate losses?
17. Which products, categories, customer segments, or regions are most negatively affected by discounting?

#### Shipping Performance
18. How long does Superstore take to ship orders?
19. How does shipping time vary by shipping mode?
20. Are there noticeable relationships between shipping mode, sales, and profitability?

#### Trends and Opportunities
21. Are there seasonal patterns in sales and profitability?
22. Are high sales consistently associated with high profits?
23. What characteristics are common among loss-making transactions?
24. Where are the greatest opportunities for Superstore to improve profitability while maintaining sustainable sales growth?

### 1.5 Key Performance Indicators (KPIs)

The following KPIs will be used to evaluate Superstore's business performance:

| KPI | Formula | Purpose |
|---|---|---|
| Total Sales | Sum of Sales | Measures total revenue generated |
| Total Profit | Sum of Profit | Measures overall profitability |
| Total Orders | Count of unique Order IDs | Measures the number of distinct customer orders |
| Total Quantity Sold | Sum of Quantity | Measures the total number of units sold |
| Average Order Value (AOV) | Total Sales / Total Orders | Measures the average revenue generated per order |
| Overall Profit Margin | Total Profit / Total Sales × 100 | Measures how much profit is generated from each unit of sales |
| Average Discount | Sum of Discount / Number of transaction lines | Measures the average discount applied across transaction lines |
| Sales Growth Rate | (Current Period Sales - Previous Period Sales) / Previous Period Sales × 100 | Measures the percentage change in sales between periods |
| Profit Growth Rate | (Current Period Profit - Previous Period Profit) / Previous Period Profit × 100 | Measures the percentage change in profit between periods |
| Average Shipping Time | Sum of Shipping Days / Number of Orders | Measures the average time between order and shipment dates |
| Loss-Making Transaction Rate | Number of loss-making transaction lines / Total transaction lines × 100 | Measures the proportion of transaction lines generating losses |

### 1.6 Decisions the Analysis Should Support

The analysis should help Superstore management make better decisions about:

- **Products:** Identify which products and categories to promote, review, or improve.
- **Discounts:** Determine where discounts may be contributing to losses and where discount policies should be reviewed.
- **Geographic Markets:** Identify high-performing and underperforming regions, states, and cities.
- **Customers:** Identify valuable customers and customer segments that contribute strongly to the business.
- **Sales Strategy:** Identify opportunities to grow sales without sacrificing profitability.
- **Shipping:** Understand shipping performance and identify possible areas for improvement.
- **Resource Allocation:** Help management focus attention and resources on the areas with the greatest potential for profitable growth.

### 1.7: Initial Dataset Inspection

In [1]:
import pandas as pd

In [5]:
df = pd.read_csv("../data/raw/Sample - Superstore.csv", encoding="latin1")

In [6]:
df.head()

,Row ID,Order ID,Order Date,Ship Date,Ship Mode,Customer ID,Customer Name,Segment,Country,City,...,Postal Code,Region,Product ID,Category,Sub-Category,Product Name,Sales,Quantity,Discount,Profit
0,1,CA-2016-152156,11/8/2016,11/11/2016,Second Class,CG-12520,Claire Gute,Consumer,United States,Henderson,...,42420,South,FUR-BO-10001798,Furniture,Bookcases,Bush Somerset Collection Bookcase,261.9600,2,0.00,41.9136
1,2,CA-2016-152156,11/8/2016,11/11/2016,Second Class,CG-12520,Claire Gute,Consumer,United States,Henderson,...,42420,South,FUR-CH-10000454,Furniture,Chairs,"Hon Deluxe Fabric Upholstered Stacking Chairs,...",731.9400,3,0.00,219.5820
2,3,CA-2016-138688,6/12/2016,6/16/2016,Second Class,DV-13045,Darrin Van Huff,Corporate,United States,Los Angeles,...,90036,West,OFF-LA-10000240,Office Supplies,Labels,Self-Adhesive Address Labels for Typewriters b...,14.6200,2,0.00,6.8714
3,4,US-2015-108966,10/11/2015,10/18/2015,Standard Class,SO-20335,Sean O'Donnell,Consumer,United States,Fort Lauderdale,...,33311,South,FUR-TA-10000577,Furniture,Tables,Bretford CR4500 Series Slim Rectangular Table,957.5775,5,0.45,-383.0310
4,5,US-2015-108966,10/11/2015,10/18/2015,Standard Class,SO-20335,Sean O'Donnell,Consumer,United States,Fort Lauderdale,...,33311,South,OFF-ST-10000760,Office Supplies,Storage,Eldon Fold 'N Roll Cart System,22.3680,2,0.20,2.5164


In [8]:
df.shape

(9994, 21)

In [9]:
df.columns

Index(['Row ID', 'Order ID', 'Order Date', 'Ship Date', 'Ship Mode',
       'Customer ID', 'Customer Name', 'Segment', 'Country', 'City', 'State',
       'Postal Code', 'Region', 'Product ID', 'Category', 'Sub-Category',
       'Product Name', 'Sales', 'Quantity', 'Discount', 'Profit'],
      dtype='str')

In [10]:
df.info()

<class 'pandas.DataFrame'>
RangeIndex: 9994 entries, 0 to 9993
Data columns (total 21 columns):
 #   Column         Non-Null Count  Dtype  
---  ------         --------------  -----  
 0   Row ID         9994 non-null   int64  
 1   Order ID       9994 non-null   str    
 2   Order Date     9994 non-null   str    
 3   Ship Date      9994 non-null   str    
 4   Ship Mode      9994 non-null   str    
 5   Customer ID    9994 non-null   str    
 6   Customer Name  9994 non-null   str    
 7   Segment        9994 non-null   str    
 8   Country        9994 non-null   str    
 9   City           9994 non-null   str    
 10  State          9994 non-null   str    
 11  Postal Code    9994 non-null   int64  
 12  Region         9994 non-null   str    
 13  Product ID     9994 non-null   str    
 14  Category       9994 non-null   str    
 15  Sub-Category   9994 non-null   str    
 16  Product Name   9994 non-null   str    
 17  Sales          9994 non-null   float64
 18  Quantity       9994

### 1.8 Data Dictionary

| Column | Description | Type |
|---|---|---|
| Row ID | Unique identifier for each dataset row | ID |
| Order ID | Unique identifier for each customer order | ID |
| Order Date | Date the customer placed the order | Date |
| Ship Date | Date the order was shipped | Date |
| Ship Mode | Shipping method used | Categorical |
| Customer ID | Unique identifier for each customer | ID |
| Customer Name | Name of the customer | Categorical |
| Segment | Customer segment | Categorical |
| Country | Country where the order was placed | Categorical |
| City | Customer/order city | Categorical |
| State | Customer/order state | Categorical |
| Postal Code | Postal code of the location | ID / Geographic |
| Region | Geographic region | Categorical |
| Product ID | Unique identifier for a product | ID |
| Category | Main product category | Categorical |
| Sub-Category | Product sub-category | Categorical |
| Product Name | Name of the product | Categorical |
| Sales | Sales revenue generated by the transaction line | Numerical |
| Quantity | Number of units sold | Numerical |
| Discount | Discount applied to the transaction line | Numerical |
| Profit | Profit generated by the transaction line | Numerical |

### 1.9 Initial Observations and Assumptions

#### Initial Observations

- The dataset contains 9,994 transaction-line records and 21 columns.
- The dataset is structured at the order-line level, meaning one order may appear across multiple rows when it contains multiple products.
- The data contains information about orders, customers, products, geographic locations, shipping, sales, discounts, quantities, and profit.
- The dataset covers multiple geographic regions and product categories.
- Sales and Profit are provided, but there is no separate cost-of-goods column.
- Order Date and Ship Date can be used to analyze sales trends and shipping times.
- The raw CSV requires explicit character encoding when loaded into Pandas.

#### Initial Assumptions

- Each Order ID represents a unique customer order.
- Each Customer ID represents a unique customer.
- Each Product ID represents a product identifier.
- Sales represents revenue generated by a transaction line.
- Profit represents the profit associated with a transaction line.
- Discount is represented as a decimal value, where 0.20 represents a 20% discount.
- Negative profit values represent loss-making transaction lines.
- The analysis will treat the provided data as historical business transaction data.
- These assumptions will be validated where possible during the data quality assessment.